File: suites/01_algorithms/comparisons/prime_sieve.c
#include <stdio.h>

static int is_prime(long long n) {
    if (n < 2) return 0;
    for (long long d = 2; d * d <= n; ++d) {
        if (n % d == 0) return 0;
    }
    return 1;
}

static long long prime_count(long long limit) {
    long long total = 0;
    for (long long n = 2; n <= limit; ++n) {
        total += is_prime(n);
    }
    return total;
}

int main(void) {
    printf("%lld\n", prime_count(25000));
    return 0;
}

File: suites/01_algorithms/comparisons/prime_sieve.rs
fn is_prime(n: i64) -> bool {
    if n < 2 {
        return false;
    }
    let mut d: i64 = 2;
    while d * d <= n {
        if n % d == 0 {
            return false;
        }
        d += 1;
    }
    true
}

fn prime_count(limit: i64) -> i64 {
    let mut total: i64 = 0;
    for n in 2..=limit {
        if is_prime(n) {
            total += 1;
        }
    }
    total
}

fn main() {
    println!("{}", prime_count(25000));
}
